imagej/imagej.py

```diff
@@ -321,6 +472,37 @@ def from_java(self, data):
                 raise exc
             return to_python(data)
 
+        def _dataset_to_xarray(self, dataset):
+            """
+            Converts an ImageJ dataset into an xarray
+            :param dataset: ImageJ dataset
+            :return: xarray with reversed (C-style) dims and coords as labeled by the dataset
+            """
+            attrs = self._ij.py.from_java(dataset.getProperties())
+            axes = [(cast('net.imagej.axis.DefaultLinearAxis', dataset.axis(idx)))
```

Comment:
Cast to the interface, `LinearAxis`, because it has all the API you need. Otherwise, any alternative implementations of `LinearAxis` will not work with this API, defeating the purpose of the interface-driven design.